tests/integration/questionnaire/test_questionnaire_save_come_back.py

```diff
@@ -0,0 +1,109 @@
+from tests.integration.create_token import create_token
+from tests.integration.integration_test_case import IntegrationTestCase
+
+
+class TestSaveComeBack(IntegrationTestCase):
+
+    def test_save_come_back_with_mandatory_question_not_answered(self):
+        # We can save and go to the sign-out page without having to fill in mandatory answer
+        base_url = '/questionnaire/1/0205/789/'
+
+        # Given
+        token = create_token('0205', '1')
+        self.client.get('/session?token=' + token.decode(), follow_redirects=False)
+
+        # When
+        post_data = {
+            'action[start_questionnaire]': 'Start Questionnaire'
+        }
+        resp = self.client.post(base_url + 'introduction', data=post_data, follow_redirects=False)
+        self.assertEquals(resp.status_code, 302)
+
+        block_one_url = resp.headers['Location']
+
+        post_data = {
+            "4b75a6f7-9774-4b2b-82dc-976561189a99": " 1000",
+            "action[save_come_back]": "Save and come back later"
+        }
+        resp = self.client.post(block_one_url, data=post_data, follow_redirects=False)
+        self.assertEquals(resp.status_code, 302)
+
+        # Then
+        # we are presented with the sign out page
+        self.assertTrue("sign-out" in resp.headers['Location'])
+        self.assertEquals(resp.status_code, 302)
+
+        resp = self.client.get(resp.headers['Location'], follow_redirects=False)
+        self.assertEquals(resp.status_code, 200)
+
+    def test_save_come_back_with_non_mandatory_validation_error(self):
+        # We can't save if a validation error is caused, this doesn't include missing a mandatory question
+        base_url = '/questionnaire/1/0205/789/'
+
+        # Given
+        token = create_token('0205', '1')
+        self.client.get('/session?token=' + token.decode(), follow_redirects=False)
+
+        # When
+        post_data = {
+            'action[start_questionnaire]': 'Start Questionnaire'
+        }
+        resp = self.client.post(base_url + 'introduction', data=post_data, follow_redirects=False)
+        self.assertEquals(resp.status_code, 302)
+
+        block_one_url = resp.headers['Location']
+
+        post_data = {
+            "4b75a6f7-9774-4b2b-82dc-976561189a99": "error",
+            "action[save_come_back]": "Save and come back later"
+        }
+        resp = self.client.post(block_one_url, data=post_data, follow_redirects=False)
+        self.assertEquals(resp.status_code, 200)
+        # Then
+        # we are presented with an error message
+        content = resp.get_data(True)
+        self.assertRegexpMatches(content, 'Please only enter whole numbers into the field.')
+
+    def test_save_come_back_complete_a_block_then_revist_it_and_use_save_and_come_back(self):
```

Comment:
Should this test be name `test_save_come_back_complete_a_block_then_revist_it` minus `_and_use_save_and_come_back`